### DISCUSSION TITLE
What is the solution for using sqflite for both android and windows?

### DISCUSSION THREAD

**Discussion by hulcon** [tekartik/sqflite#784] [Q&A] [answered]
One of my flutter projects needs to be deployed on Android and Windows. The app needs to store a lot of relational data on local storage. The documentation says that it supports only Android and iOS while as support for desktop platforms is achieved through sqflite_common_ffi. So what do i need to do if i need support for both Android and Windows? Do i need to include both of those packages (sqflite and sqflite_common_ffi) or what? I am a bit familiar with SQFLite as i have previously used it in one of my projects but i haven't used the sqflite_common_ffi yet. Please suggest the procedure to achieve functionality on both platforms.

**Comment by alextekartik** [comment, marked as answer]
Sorry for the lack of documentation. Indeed you can add both. (it will use sqflite on iOS/Android and sqlite_common_ffi in Windows). The API is the same. You only need to provide the proper factory on start such as:

```dart
import 'package:sqflite_common_ffi/sqflite_ffi.dart';
import 'package:sqflite/sqflite.dart';

Future main() async {
  if (Platform.isWindows || Platform.isLinux) {
    // Initialize FFI and use it on Windows and linux
    sqfliteFfiInit();
    databaseFactory = databaseFactoryFfi;
  }
  // Use the default factory on iOS/Android
  runApp(MyApp());
}
```

As a a side note, don't rely don't `getDatabasesPath()` on Windows, use `path_provider` instead to find a proper database file location.

More information [here](https://github.com/tekartik/sqflite/blob/master/sqflite_common_ffi/doc/using_ffi_instead_of_sqflite.md)
You can also use sqflite_common_ffi on both iOS/Android and Windows!

**Reply by hulcon** [reply]
Thank you for the reply but can you please elaborate a bit on your side note "don't rely don't getDatabasesPath() on Windows, use path_provider instead to find a proper database file location." 
Does it mean that i should, for example, use the getApplicationDocumentsDirectory() method of path_provider for Windows?

**Comment by alextekartik** [comment]
> Does it mean that i should, for example, use the getApplicationDocumentsDirectory() method of path_provider for Windows?

Correct. Or user whatever local folder you want to save your database to (absolute or relative to your app or your current directory). By default getDatabasePaths returns a dummy local path on Windows.

**Reply by hulcon** [reply]
Thank you very much for your quick response.
